arlo_server/audit_boards.py

```diff
@@ -186,6 +187,70 @@ def list_audit_boards(
     return jsonify({"auditBoards": json_audit_boards})
 
 
+MEMBER_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "name": {"type": "string"},
+        "affiliation": {
+            "anyOf": [
+                {
+                    "type": "string",
+                    "enum": [affiliation.value for affiliation in Affiliation],
+                },
+                {"type": "null"},
+            ]
+        },
+    },
+    "additionalProperties": False,
+    "required": ["name", "affiliation"],
+}
+
+SET_MEMBERS_SCHEMA = {
+    "type": "array",
+    "items": MEMBER_SCHEMA,
+}
+
+
+def validate_members(members: List[JSONDict]):
+    # You can do all of these checks using JSON schema, but the resulting error
+    # messages aren't very good.
+    if len(members) == 0:
+        raise BadRequest("Must have at least one member.")
+    if len(members) > 2:
```

Comment:
It's a bit odd that the check for members length is abstracted here while the code further down that calls into this check also assumes there are 1 or 2 members. Maybe a slight violation of DRY? I wonder if the length check should be in the set_audit_board_members function, in case we ever change it, that's where it belongs.

Reply:
Hmm, I've been putting validation in its own function on every endpoint. But I am curious to understand your comment more deeply. Let's discuss